EIPS/eip-3155.md

```diff
@@ -0,0 +1,259 @@
+
+Optional Fields: 
+| Name  | Type  | Explanation  |
+|---|---|---|
+| `opName` | String  | Name of the operation  |
+| `error`  | Hex-String  | Description of an error (should contain revert reason if supported)  |
+| `memory` | Array of Hex-Strings  | Array of all allocated values  |
+| `storage` |  Key-Value |  Array of all stored values |
+
+*Example:* 
+```
+{"pc":0,"op":96,"gas":"0x2540be400","gasCost":"0x3","memory":"0x","memSize":0,"stack":[],"depth":1,"error":null,"opName":"PUSH1"}
+```
+
+The `stack`, `memory` and `memSize` are the values _before_ execution of the op. 
+All array attributes (`stack`, `returnStack`, `memory`) MUST be initialized to empty arrays ("stack":[],) NOT to null.
+If the CUT will not output values for `memory` or `storage` then the `memory` and `storage` fields are omitted.
+This can happen either because the CUT does not support tracing these fields or it has been configured not to trace it.
+The `memSize` field MUST be present regardless of `memory` support.
+Clients SHOULD implement a way to disable recording the storage as the stateroot includes all storage updates.
+Clients SHOULD output the fields in the same order as listed in this EIP.
+
+The CUT MUST NOT output a line for the `STOP` operation if an error occurred:
+*Example:* 
+```
+{"pc":2,"op":0,"gas":"0x2540be3fd","gasCost":"0x0","memory":"0x","memSize":0,"stack":["0x40"],"depth":1,"error":null,"opName":"STOP"}
+```
+
+### Summary and error handling
+
+At the end of execution, the CUT MUST print some summerical info, this info SHOULD have the following fields. The summary SHOULD be printed to STDOUT.
+The summary should be a single jsonl object.
+
+Required Fields: 
+| Name  | Type  | Explanation  |
+|---|---|---|
+| `stateRoot` | Hex-String  | Root of the state trie after executing the transaction  |
+| `output` |   | Return values of the function  |
+| `gasUsed` | Hex-Number  | All gas used by the transaction  |
+| `pass` | Boolean  | Bool whether transaction was executed successfully |
+
+OptionalFields:
+| Name  | Type  | Explanation  |
+|---|---|---|
+| `time` |  Number | Time in nanoseconds needed to execute the transaction  |
+| `fork` | String  | Name of the fork rules used for execution  |
+```
+{"stateRoot":"0xd4c577737f5d20207d338c360c42d3af78de54812720e3339f7b27293ef195b7","output":"","gasUsed":"0x3","successful":"true","time":141485}
+```
+
+The state test format allows for multiple `postStates` for different forks.
+If different `postStates` are present the CUT MUST execute the test according to the rules of the fork sequentially.
+It MUST print a trace for each execution as well as an individual summary for each test.
+
+
+## Rationale
+This EIP is created to enable easier testing of EVM implementations to create a safer ecosystem. It is largely based on the previous non-official documentation for EVM tracing. It tries to cover as many corner cases as possible to enable true client compatibility.
```

Comment:
This all sounds like motivation, not Rationale.  Rationale is a place to discuss why specific decisions were made.  Why was a particular name used instead of another, why was a particular constant used instead of another, why was a particular design chosen instead of another, etc.